Formula/nginx-unit.rb

```diff
@@ -0,0 +1,223 @@
+    EOS
+  end
+
+  plist_options :manual => "#{HOMEBREW_PREFIX}/sbin/unitd --no-daemon"
+
+  def plist; <<~EOS
+    <?xml version="1.0" encoding="UTF-8"?>
+    <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+    <plist version="1.0">
+      <dict>
+        <key>Label</key>
+        <string>#{plist_name}</string>
+        <key>RunAtLoad</key>
+        <true/>
+        <key>KeepAlive</key>
+        <false/>
+        <key>ProgramArguments</key>
+        <array>
+            <string>#{opt_sbin}/unitd</string>
+            <string>--no-daemon</string>
+        </array>
+        <key>WorkingDirectory</key>
+        <string>#{HOMEBREW_PREFIX}</string>
+      </dict>
+    </plist>
+    EOS
+  end
+
+  test do
+    require "json"
+    require "net/http"
+    require "socket"
+    require "timeout"
+
+    begin
+      # We need to install rack for the ruby module test to work, so set a
+      # temporary gems path before booting unitd. (unitd boots and functions
+      # fine without, but fails to start an app until rack is installed.)
+      # And we install old rack for el capitan / sierra compat.
+      rubygems = testpath/"ruby/gems"
+      system "gem", "install", "rack", "-v", "1.6.10", "--install-dir=#{rubygems}"
+      ENV["GEM_PATH"] = rubygems
+
+      control = testpath/"unit.control.sock"
+
+      pid = spawn "#{sbin}/unitd", "--no-daemon", "--log", "/dev/stdout", "--control", "unix:#{control}"
+      timeout(5) { sleep 0.1 until control.exist? }
+
+      # Unit uses a unix control socket. Net::HTTP to a socket is hard. Sorry.
+      socket = Net::BufferedIO.new(UNIXSocket.new(control))
+      request = Net::HTTP::Get.new("/")
+      request.exec(socket, "1.1", request.path)
+      response = Net::HTTPResponse.read_new(socket)
+      response.reading_body(socket, request.response_body_permitted?) {}
+
+      assert_equal("200", response.code)
+      expected_body = { "listeners" => {}, "applications" => {} }
+      assert_equal expected_body, JSON.parse(response.body)
+
+      ### Perl module
+
+      socket = Net::BufferedIO.new(UNIXSocket.new(control))
+      request = Net::HTTP::Put.new("/config")
+      request.body = JSON.generate({})
+      request.exec(socket, "1.1", request.path)
+      response = Net::HTTPResponse.read_new(socket)
+      response.reading_body(socket, request.response_body_permitted?) {}
+
+      perlapp = testpath/"perl/app.psgi"
+      perlapp.write <<~EOS
+        my $app = sub {
+          my $env = shift;
+          return [ '200', [ 'Content-Type' => 'text/plain' ], [ "Hello, perl world!" ] ];
+        }
+      EOS
+
+      socket = Net::BufferedIO.new(UNIXSocket.new(control))
+      request = Net::HTTP::Put.new("/config")
+      request.body = JSON.generate(
+        "listeners" => { "*:8000" => { "application" => "perlapp" } },
+        "applications" => { "perlapp" => { "type" => "perl", "working_directory" => perlapp.parent, "script" => perlapp } },
```

Comment:
tweak the formatting to avoid going beyond the 80th column please, here and below

Reply:
Done!